python_modules/dagster/dagster/_core/execution/with_resources.py

```diff
@@ -96,6 +97,9 @@ def asset2(context):
                 )
             resource_defs[key] = resource_defs[key].configured(resource_config["config"])
 
+    if not isinstance(definitions, IterableABC):
+        definitions = [definitions]
```

Comment:
Nit: prefer to check whether it is directly an instance of the classes we expect if possible, otherwise introduces a potential footgun with bad error message?